modules/debugging/overview.md

```diff
@@ -1,23 +1,229 @@
-approach: understand the system.
-read the manual.
-know what's reasonable 
-know some of the fundamentals of your technical field.
-have a high-level view of your system; know the components and their interfaces 
-know your tools.
-approach: make it fail.
-repro steps that start from the beginning.
-*stimulate* the failure (spray a hose on a leaky window).
-approach: quit thinking and look
-instrument the system ahead of time.
-instrument the system when you need it.
-see the failure.
-see the details.
-guess only to focus the search.
-approach: divide and conquer.
-approach: change one thing at a time.
-approach: keep an audit trail.
-approach: check the plug.
-get a fresh view.
-approach: if you didn't fix it, it ain't fixed.
-
-https://www.amazon.com/Debugging-Indispensable-Software-Hardware-Problems/dp/0814471684
\ No newline at end of file
+# Debugging Module
+
+Debugging WILL be the most time consuming process on your job. 
+
+Everyone will do it. Nobody can avoid it. Nobody's code works the first time.
+
+So how do you approach it?
+
+There are many different approaches
+
+### Approach 1: Understand the system.
+
+The approach of always understanding everything before you debug. While it will be
+easier to debug like this, it is also time consuming and hard to know most aspects of
+what you are doing.
+
+1. Read the manual
+2. Know what's reasonable
+3. Know some of the fundamentals of your technical field.
+4. Have a high-level view of your system; Know the components and their interfaces.
+5. Know your tools.
+
+### Approach 2: Make it fail.
+
+The simplest and arguably most common approach.
+
+1. Repro steps that start from the beginning.
+2. *stimulate* the failure (spray a hose on a leaky window).
+
+### Approach 3: Quit thinking and look.
+
+Sometimes the error is very obvious by just reading through your code again.
+
+1. Instrument the system ahead of time.
+2. Instrument the system when you need it.
+3. See the failure.
+4. See the details.
+5. Guess only to focus the search.
+
+### Approach 4: Divide and conquer.
+
+If you divide your code into multiple parts and make sure each works. Together
+the usually will as well. 
+
+1. Divide code into large parts.
+2. Make the parts smaller and smaller until the error is on one small portion.
+3. Use any other approach to fix this error. 
+
+<!-- ### Approach 5: Change one thing at a time.  FEEL LIKE THIS IS SORT OF A BAD DEBUGGING PROCESS ON ITS OWN-->
```

Comment:
Not sure what you mean.  First, I don't think the idea is that each of the bullet points is meant to represent an entire process that's used in isolation.

And changing one thing at a time is part of the essence of the scientific method, wouldn't you say.  To establish or at least increase the probability of clear causation, isolate one variable.